pkg/k8s/apis/cilium.io/v2/register_test.go

```diff
@@ -77,3 +78,29 @@ func (s *CiliumV2RegisterSuite) TestNeedsUpdateCorruptedVersion(c *C) {
 
 	c.Assert(needsUpdate(crd), Equals, true)
 }
+
+func (s *CiliumV2RegisterSuite) TestCIDRRegex(c *C) {
+	testCIDRs := []string{
+		"192.0.2.3/32",
+		"192.0.2.0/24",
+		"0.0.0.0/0",
+		"::/0",
+		"::cafe/128",
+		"f00d::/64",
+		"f00d::0:0/120",
+	}
+
+continueTest:
+	for _, input := range testCIDRs {
+		for _, prop := range CIDR.OneOf {
+			matched, err := regexp.MatchString(prop.Pattern, input)
+			c.Assert(err, IsNil)
+			if matched {
+				continue continueTest
+			}
+		}
+		// The below is always false, we shouldn't hit this with valid
+		// CIDR prefixes.
+		c.Assert(input, Equals, "failed to match CIDR.OneOf[*].Pattern")
```

Comment:
This assert is fatal to the test, right? From the code structure, it looks like you intended to continue the outer loop, though.

Reply:
I added negative tests.